File: columnwise_corrcoef_perf.jl
# This is columnwise correlation in Julia, to compare with numpy.
# Julia designers were smart enough to implement the built-in cor function
# that computes correaltion columnwise.
#
# Ilya Kizhvatov, with help from Cees-Bart Breunesse

# system information
versioninfo()

# dummy arrays for the dry run
foo = rand(Float64, 100000, 1000)
bar = rand(Float64, 100000, 256)

# arrays representing real data
O = rand(Float64, 100000, 1000)
P = rand(Float64, 100000, 256)

# timing dry run to compile the function and the timing functions
println("Dry run...")
@time cor(foo, bar)

# real timing
println("Real run...")
@time cor(O, P)

File: columnwise_corrcoef_perf.m
% Performance of column-wise correlation coefficient
%
% Requires Statistic and Machine Learning Toolbox
%
% Ilya Kizhvatov

% dummy arrays for the dry run
foo = rand(100000, 1000);
bar = rand(100000, 256);

% arrays representing real data
O = randn(100000, 1000);
P = randn(100000, 256);

% dry run
disp('Dry run...')
tic
corr(foo, bar);
toc

% real run
disp('Real run...')
tic
corr(O, P);
toc